### PR TITLE
Feature/nep 120 optimize imports

### PR DESCRIPTION
**Pull request by OleksandrOpanasiuk**

### CODE REVIEW

File: src/main/java/com/parsing/parsers/pdf/parsing/ManyTableDetector.java [hunk 1]
```diff
@@ -61,12 +61,10 @@ private int groupVerticalLines(List<double[]> lines, List<Double> mergedY) {
         for (Double yUp : yUpList) {
             for (Double yDown : yDownList) {
                 if (yDown < yUp) {
-                    log.debug("Ydown " + yDown + " Yup" + yUp);
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/ManyTableDetector.java:69]: I think that we need to consider using debug log level for this type of exception. And we can use INFO for logs that help us to follow the program flow. Do you need it at all? Or we can consider using not Slf4j but custom logs like in the correct answer here: https://stackoverflow.com/questions/12089902/custom-log-level so essentially you can set this logger in the class: 
private static Logger parserLogger = Logger.getLogger(MyClass.class) 
But we need to check that it doesn't work when we have a DEBUG or INFO level. We need an ability to turn it on in application.yml
So perhaps we will need to do it with Log4j like here: https://medium.com/codex/log4j2-create-custom-log-levels-and-how-to-use-them-48685e133fd1 but it's another task I suppose

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java [hunk 2]
```diff
@@ -38,26 +41,40 @@ public class PageOCRPreparator {
...
     private static final String REGEX = ".*[А-ЩЬЮЯЄЇІа-щьюяєїі].*";
     private static final double[] RGB_WHITE_COLOUR = {255, 255, 255};
 
-    public String preparePage(PDDocument document, int page) throws IOException {
-        log.debug("Class PageOCRPreparator started.");
-        String fileResult = "preparePage" + page + ".png";
-        String pagePDF = getPagePDF(document, page);
-        String rotatedPage = rotationImage.rotate(pagePDF);
-        Rectangle pageRectangle = getPageMainRectangle(document, page);
+    static final double HORIZONTAL_LINE_LENGTH = 700;
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: consider making it 30% of the total horizontal size, For all project

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java [hunk 3]
```diff
@@ -38,26 +41,40 @@ public class PageOCRPreparator {
...
+    static final int HORIZONTAL_THRESHOLD = 5;
+    static final double ALL_LINES_THRESHOLD1 = 50;
+    static final double ALL_LINES_THRESHOLD2 = 200;
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java:48]: what difference between 1 and 2?

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java [hunk 4]
```diff
@@ -38,26 +41,40 @@ public class PageOCRPreparator {
...
+    static final int ALL_LINES_APERTURESIZE = 3;
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: APERTURE_SIZE ?

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java [hunk 5]
```diff
@@ -38,26 +41,40 @@ public class PageOCRPreparator {
...
+    static final int ALL_LINES_RHO = 3;
+    static final int ALL_LINES_MAXLINEGAP = 5;
+    static final double VERTICAL_LINE_LENGTH = 100;
+
+    public String preparePage(PDDocument document, int pageNumber) throws IOException {
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java:53]: use try-catch-finally and close the session in finaly block.

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java [hunk 6]
```diff
@@ -38,26 +41,40 @@ public class PageOCRPreparator {
...
-        Imgcodecs.imwrite(page + ".png", matPage);
-        java.util.List<Rectangle> pageRectanglesAllWords = getPageRectanglesAllWords(page + ".png");
-        List<Rectangle> rectanglesWithSymbols = extractTextFromRectangle(page + ".png", pageRectanglesAllWords);
-        saveRectanglesOnImage(rectanglesWithSymbols, page + ".png");
-        final Rectangle mainPageRectangle = findMainPageRectangle(rectanglesWithSymbols);
-        final Mat result = cleanTableBorders(matPage, mainPageRectangle);
+        final Mat result = cleanTableBorders(tableMat, mainPageRectangle, pageNumber);
         Imgcodecs.imwrite(fileResult, result);
-        log.debug("Class PageOCRPreparator finished.");
+        log.debug("Method PageOCRPreparator finished.");
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: Consider using INFO level to track level for such cases. And add the name of the document and a page

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java [hunk 7]
```diff
@@ -165,26 +178,27 @@ private Rectangle findMainPageRectangle(List<Rectangle> rectangles) {
                     yMax = yCurrent;
                 }
             }
-            return createRect(xMin, yMin, xMax, yMin, yMax);
+            return createRect(xMin, yMin, xMax, yMin, yMax, 5);
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: magic number

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java [hunk 8]
```diff
@@ -165,26 +178,27 @@ private Rectangle findMainPageRectangle(List<Rectangle> rectangles) {
...
         List<Rectangle> result = new ArrayList<>();
         ITesseract itesseract = new Tesseract();
         itesseract.setDatapath(getTessDataPath());
         itesseract.setLanguage("ukr+eng");
         BufferedImage bim = null;
         try {
-            log.debug(getProjectPath() + filename);
-            BufferedImage buf = ImageIO.read(new File(getProjectPath() + "\\" + filename));
+            log.debug(getProjectPath() + "/" + filename);
+            BufferedImage buf = ImageIO.read(new File(getProjectPath() + "/" + filename));
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: need to assign it to the System OS Linux or Windows

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java [hunk 9]
```diff
@@ -251,18 +266,95 @@ private String saveRectanglesOnImage(List<Rectangle> rectangles, String filename
...
+        List<double[]> lines = new ArrayList<>();
+        for (int x = 0; x < linesP.rows(); x++) {
+            double[] l = linesP.get(x, 0);
+            lines.add(l);
+        }
+        return lines;
+    }
+
+    private List<Rect> extractListRect(String filesource) {
+        double WIDTH_PERCENT = 0.15;
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: TEXT_PARSING_WIDTH_PERCENT? Consider something more informative please. Consider adding image or width of what is it.

File: src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java [hunk 10]
```diff
@@ -46,78 +47,85 @@ public class ParserPDF {
...
+                    obj.put("fileName", file.getOriginalFilename());
+                    StringBuilder builder = new StringBuilder();
+                    for (Row row : table) {
+                        for (Column column : row.getColumns()) {
+                            builder.append(column.getParsingResult());
+                        }
                     }
+                    obj.put("text", builder);
+                } else {
+                    log.debug("Table did not found on page " + page);
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java]: warn level

File: src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java [hunk 11]
```diff
@@ -46,78 +47,85 @@ public class ParserPDF {
...
                 }
-                obj.put("text", builder);
-            } else {
-                log.debug("Table did not found on page " + page);
+            } catch (RuntimeException ex) {
+                ex.printStackTrace();
+                ex.getCause();
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java]: add a dedicated exception for it with a cause and a message

File: src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java [hunk 12]
```diff
@@ -46,78 +47,85 @@ public class ParserPDF {
...
-                    String fileTableName = tableRecognizer.detectTable(prePage);
-                    table = rectangleDetector.detectRectangles(fileTableName);
+                    rectangleDetector.saveIMageWithVerticalLines2(tablesLines, page);
+                    String fileTableName = tableRecognizer.detectTable(prePage, page);
+                    table = rectangleDetector.detectRectangles(fileTableName, page);
                     table = extractTextFromScannedDocument(fileTableName);
                     boolean isRecognized = dataRecognizer.recognizeLotPDFResult(table);
                     if (isRecognized) {
-                        log.debug("Class ParserPDF.parseProzorroFileForScheduler finished");
+                        log.debug("Method parseProzorroFileForScheduler finished successful");
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java:104]: info level

File: src/main/java/com/parsing/parsers/pdf/parsing/RectangleDetector.java [hunk 13]
```diff
@@ -211,26 +212,35 @@ public List<double[]> findVerticalLinesWithOpenCV(String fileSource) {
...
+                        RectangleDetector.HORIZONTAL_LINE_LENGTH, ALL_LINES_MAXLINEGAP);
+                List<double[]> lines = new ArrayList<>();
+                for (int x = 0; x < linesP.rows(); x++) {
+                    double[] l = linesP.get(x, 0);
+                    lines.add(l);
+                }
+                return lines;
+            }
+        } else {
+            log.debug("Filesource is null.");
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/RectangleDetector.java]: add a name of a file or something that will be useful

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java [hunk 14]
```diff
@@ -19,89 +20,111 @@
...
-        double width = Math.sqrt(Math.pow((x2 - x1), POWER_TWO) + Math.pow((y2 - y1), POWER_TWO));
+        double width = Math.sqrt(Math.pow((x2 - x1), POWER_TWO));
         double height = Math.sqrt(Math.pow((y3 - y2), POWER_TWO));
-        return new Rect((int) x1, (int) y3, (int) width, (int) height);
+        return new Rect((int) x1, (int) y1, (int) width, (int) height);
+    }
+
+    public Rectangle createRectangle(double xLeftBottom, double yLeftBottom, double xRightBottom, double yRightBottom, double yRightTop) {
+        log.debug("Method createRectangle started.");
+        double width = Math.sqrt(Math.pow((xRightBottom - xLeftBottom), 2) + Math.pow((yRightBottom - yLeftBottom), 2));
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java]: magic numbers

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java [hunk 15]
```diff
@@ -19,89 +20,111 @@
...
-                x1 = l[0];
+            if (l[3] > y3) {
+                y3 = l[3];
             }
+
         }
-        Imgcodecs.imwrite("afterHough.png", cdstP);
-        Rect rectangle = createRect(x1, y1, x2, y2, y3);
+        Imgcodecs.imwrite(page + "_#8_afterHough.png", cdstP);
+        log.debug("x1 = {}; y1 = {}; x2 = {}; y2={}, y3={}", x1, y1, x2, y2, y3);
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java:108]: INfo level

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java [hunk 16]
```diff
@@ -19,89 +20,111 @@
...
-            return fileResult;
+                Rect rectNew = isImageConsistRectangle(rect, source);
+                cropTable = new Mat(source, rectNew);
+                String fileResult = page + "_#10_destination.png";
+                Imgcodecs.imwrite(fileResult, cropTable);
+                log.debug("Method detectTable finished.");
+                return fileResult;
         } catch (Exception ex) {
-            log.warn("Can't crop image. Rectangle x = " + rectangle.x + " y = " + rectangle.y + " height = " + rectangle.height
-                    + " width = " + rectangle.width + " source size = " + source.size(),  ex.getCause());
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java:108]: redundant space

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java (hunk 2 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: Done!

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java (hunk 7 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java (hunk 9 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java (hunk 10 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/RectangleDetector.java (hunk 13 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/RectangleDetector.java]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java (hunk 16 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java:108]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java (hunk 3 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java:48]: It's from opencv library, Canny's recommendation.
Params:
 threshold1 – first threshold for the hysteresis procedure.
threshold2 – second threshold for the hysteresis procedure.

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java (hunk 15 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java:108]: I think it's debug level.
INFO
INFO messages are like the normal behavior of applications. They state what happened. For example, if a particular service stopped or started or you added something to the database. These entries are nothing to worry about during usual operations. The information logged using the INFO log is usually informative, and it does not necessarily require you to follow up on it.

DEBUG
With DEBUG, you are giving diagnostic information in a detailed manner. It is verbose and has more information than you would need when using the application. DEBUG logging level is used to fetch information needed to diagnose, troubleshoot, or test an application. This ensures a smooth running application.

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java (hunk 14 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java (hunk 11 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java (hunk 6 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java (hunk 5 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java:53]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java [hunk 17]
```diff
@@ -251,18 +267,94 @@ private String saveRectanglesOnImage(List<Rectangle> rectangles, String filename
...
+
+    private boolean isTargetRect(Rect rectangle, int row, int column) {
+        double yLeftUp = rectangle.y;
+        double xLeftUp = rectangle.x;
+        double xRightDown = xLeftUp + rectangle.width;
+        double yRightDown = yLeftUp + rectangle.height;
+        return xLeftUp <= column && column <= xRightDown && yLeftUp <= row && row <= yRightDown;
+    }
+
+    private double getOffset(Rectangle rectangle, List<double[]> list) {
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java:299]: reconsider name list. maybe it's lines

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java [hunk 18]
```diff
@@ -251,18 +267,94 @@ private String saveRectanglesOnImage(List<Rectangle> rectangles, String filename
...
+        double rightOffset;
+        double leftOffset;
+        double leftRectangleX = rectangle.getX();
+        double rightRectangeX = rectangle.getX() + rectangle.width;
+        double rightListX = list.get(0)[0];
+        double leftListX = list.get(0)[2];
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: add X1 and Y2 (or something like this) constants

File: src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java [hunk 19]
```diff
@@ -46,78 +48,84 @@ public class ParserPDF {
...
-        if (!pdfDir.exists()) {
-            pdfDir.mkdir();
-        }
-        return currentPathPosition.toAbsolutePath() + DIR_TO_READ_TESSDATA;
+            Path currentPathPosition = Paths.get("").toAbsolutePath();
+            File pdfDir = new File(currentPathPosition + DIR_TO_READ_TESSDATA);
+            if (!pdfDir.exists()) {
+                pdfDir.mkdir();
+            }
+            return currentPathPosition.toAbsolutePath() + DIR_TO_READ_TESSDATA;
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java]: auto format

File: src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java [hunk 20]
```diff
@@ -1,12 +1,14 @@
...
 
+import com.parsing.exception.ParseProzorroFileException;
 import com.parsing.parsers.pdf.parsing.model.Column;
 import com.parsing.parsers.pdf.parsing.model.Row;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import net.sourceforge.tess4j.ITesseract;
 import net.sourceforge.tess4j.Tesseract;
 import net.sourceforge.tess4j.TesseractException;
+import net.sourceforge.tess4j.util.LoadLibs;
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java:11]: do you use it?

File: src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java [hunk 21]
```diff
@@ -141,7 +149,8 @@ private List<Row> extractTextFromScannedDocument(String fileTableName) {
                 log.debug(filename + " = " + result);
             }
         }
-        return table;
+        return table;}
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java]: consider moving down the curly bracket.

File: src/main/java/com/parsing/parsers/pdf/parsing/RotationImage.java [hunk 22]
```diff
@@ -14,14 +14,14 @@
 @Slf4j
 public class RotationImage {
 
-    public String rotate(String filename) throws IOException {
-        log.debug("Class RotationImage started.");
-        String result = "rotatedImage.jpg";
+    public String rotateImage(String filename, int pageNumber) throws IOException {
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/RotationImage.java:17]: I suppose that Yura is doing fixing IOExceptions, look in his PR please

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java [hunk 23]
```diff
@@ -11,6 +11,7 @@
 import org.opencv.imgproc.Imgproc;
 import org.springframework.stereotype.Component;
 
+import java.awt.*;
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java]: don't use asterisks

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java [hunk 24]
```diff
@@ -24,89 +25,111 @@
...
-    private static double x1 = 500;
-    private static double y1 = 0;
-    private static double x2 = 0;
-    private static double y2 = 0;
-    private static double y3 = 1500;
+    private static double x1, y1, x2, y2, y3;
     private static int POWER_TWO = 2;
 
     private static Rect createRect(double x1, double y1, double x2, double y2, double y3) {
-        double width = Math.sqrt(Math.pow((x2 - x1), POWER_TWO) + Math.pow((y2 - y1), POWER_TWO));
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java:36]: check it please, should it be deleted?

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java [hunk 25]
```diff
@@ -24,89 +25,111 @@
...
         Mat source = Imgcodecs.imread(fileSource, Imgcodecs.IMREAD_GRAYSCALE);
         Imgproc.Canny(source, dst, 50, 200, 3, false);
-        Imgcodecs.imwrite("afterCanny.png", dst);
+        Imgcodecs.imwrite(page + "_#7_afterCanny.png", dst);
         Imgproc.cvtColor(dst, cdst, Imgproc.COLOR_GRAY2BGR);
         cdstP = cdst.clone();
         Mat linesP = new Mat();
         Imgproc.HoughLinesP(dst, linesP, 1, Math.PI / 180, 25, 700, 10);
+        double[] array = linesP.get(0, 0);
+        x1 = array[0];
```
Comment (FedosovMax) [inline src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java]: consider adding constants

**Review by FedosovMax — APPROVED** [review]

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java (hunk 4 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java (hunk 17 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java:299]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java (hunk 18 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java (hunk 20 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java:11]: No, deleted. Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java (hunk 21 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/ParserPDF.java]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java (hunk 24 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java:36]: Yes.

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java (hunk 25 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java]: Done.

File: src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java (hunk 23 above)
Comment (OleksandrOpanasiuk) [reply src/main/java/com/parsing/parsers/pdf/parsing/TableRecognizer.java]: Done.

**Pull request merged**